### ISSUE TITLE
Can someone please explain how to setup config.py? My training is getting cancelled by itself

### ISSUE THREAD

**Issue by turdubars** [deepinsight/insightface#1203]
I'm trying to run training on **Colab** to check the training process, but it is getting cancelled while the training with default configs on Casia-webface dataset

**default config.py is used, I only added this part for dataset:**

dataset.casia = edict()
dataset.casia.dataset = 'casia'
dataset.casia.dataset_path = '/content/drive/My Drive/$CASIA_WEBFACE_ROOT'
dataset.casia.num_classes = 10572
dataset.casia.image_shape = (112,112,3)

**Command that I'm running**: *CUDA_VISIBLE_DEVICES='0' python -u train.py --network r100 --loss arcface --dataset casia*

**Output from the shell:**

gpu num: 1
prefix ./models/r100-arcface-casia/model
image_size [112, 112]
num_classes 10572
Called with argument: Namespace(batch_size=128, ckpt=3, ctx_num=1, dataset='casia', frequent=20, image_channel=3, kvstore='device', loss='arcface', lr=0.1, lr_steps='100000,160000,220000', models_root='./models', mom=0.9, network='r100', per_batch_size=128, pretrained='', pretrained_epoch=1, rescale_threshold=0, verbose=2000, wd=0.0005) {'bn_mom': 0.9, 'workspace': 256, 'emb_size': 512, 'ckpt_embedding': True, 'net_se': 0, 'net_act': 'prelu', 'net_unit': 3, 'net_input': 1, 'net_blocks': [1, 4, 6, 2], 'net_output': 'E', 'net_multiplier': 1.0, 'val_targets': ['lfw', 'cfp_fp', 'agedb_30'], 'ce_loss': True, 'fc7_lr_mult': 1.0, 'fc7_wd_mult': 1.0, 'fc7_no_bias': False, 'max_steps': 0, 'data_rand_mirror': True, 'data_cutoff': False, 'data_color': 0, 'data_images_filter': 0, 'count_flops': True, 'memonger': False, 'loss_name': 'margin_softmax', 'loss_s': 64.0, 'loss_m1': 1.0, 'loss_m2': 0.5, 'loss_m3': 0.0, 'net_name': 'fresnet', 'num_layers': 100, 'dataset': 'casia', 'dataset_path': '/content/drive/My Drive/Deep Learning/Datasets/casia-webface', 'num_classes': 10572, 'image_shape': [112, 112, 3], 'loss': 'arcface', 'network': 'r100', 'num_workers': 1, 'batch_size': 128, 'per_batch_size': 128}
0 1 E 3 prelu False
Network FLOPs: 24.2G
INFO:root:loading recordio /content/drive/My Drive/Deep Learning/Datasets/casia-webface/train.rec...
header0 label [490624. 501196.]
id2range 10572
490623
rand_mirror True
tcmalloc: large alloc 1806336000 bytes == 0xa650000 @  0x7efd7b731b6b 0x7efd7b751379 0x7efd36eaae2a 0x7efd36eb3f0f 0x7efd3657a6d0 0x7efd3657b19d 0x7efd7320adae 0x7efd7320a71f 0x7efd7341e5c4 0x7efd7341ec33 0x5a9eec 0x50a783 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x509918 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d
tcmalloc: large alloc 1806336000 bytes == 0x760f8000 @  0x7efd7b731b6b 0x7efd7b751379 0x7efd36eaae2a 0x7efd36eb3f0f 0x7efd3657a6d0 0x7efd3657b19d 0x7efd7320adae 0x7efd7320a71f 0x7efd7341e5c4 0x7efd7341ec33 0x5a9eec 0x50a783 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x509918 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d
loading bin 0
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
(12000, 3, 112, 112)
ver lfw
tcmalloc: large alloc 2107392000 bytes == 0xe3366000 @  0x7efd7b731b6b 0x7efd7b751379 0x7efd36eaae2a 0x7efd36eb3f0f 0x7efd3657a6d0 0x7efd3657b19d 0x7efd7320adae 0x7efd7320a71f 0x7efd7341e5c4 0x7efd7341ec33 0x5a9eec 0x50a783 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x509918 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d
tcmalloc: large alloc 2107392000 bytes == 0x160d2a000 @  0x7efd7b731b6b 0x7efd7b751379 0x7efd36eaae2a 0x7efd36eb3f0f 0x7efd3657a6d0 0x7efd3657b19d 0x7efd7320adae 0x7efd7320a71f 0x7efd7341e5c4 0x7efd7341ec33 0x5a9eec 0x50a783 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x509918 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d
loading bin 0
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
loading bin 12000
loading bin 13000
(14000, 3, 112, 112)
ver cfp_fp
tcmalloc: large alloc 1806336000 bytes == 0x1df2b0000 @  0x7efd7b731b6b 0x7efd7b751379 0x7efd36eaae2a 0x7efd36eb3f0f 0x7efd3657a6d0 0x7efd3657b19d 0x7efd7320adae 0x7efd7320a71f 0x7efd7341e5c4 0x7efd7341ec33 0x5a9eec 0x50a783 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d 0x50c1f4 0x509918 0x50a64d 0x50c1f4 0x507f24 0x509c50 0x50a64d
loading bin 0
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
(12000, 3, 112, 112)
ver agedb_30
lr_steps [100000, 160000, 220000]
**call reset()**
**^C**

**Comment by mlourencoeb** [comment]
What is the gpu on your Colab and which mxnet version are you using? Sounds like a memory allocation issue.

**Comment by turdubars** [comment]
> What is the gpu on your Colab and which mxnet version are you using? Sounds like a memory allocation issue.

**GPU: Tesla K80**
![Screenshot from 2020-08-13 12-57-59](https://user-images.githubusercontent.com/41260435/90103668-b9174700-dd64-11ea-8bda-3d4f0860cc5a.png)

MXNet version: 1.5.1
CUDA version: 10.1

**Comment by mlourencoeb** [comment]
Try setting --per_batch_size 16 and increase it gradually by a factor of x2 until you see the error again.

**Comment by turdubars** [comment]
> Try setting --per_batch_size 16 and increase it gradually by a factor of x2 until you see the error again.

Tried to run with different batch sizes including 8, 16 but having the same problem, it's eating too much RAM. Tried to run the training on mobilenet, it also didn't help. Tried to run on Colab, which has 11 GB of RAM

**Comment by turdubars** [comment]
The problem was with the loading of validation sets, they take too much memory in RAM (only lfw takes ~6 GB).
